File: Data Structures & Algorithms/string-encode-and-decode/submission-4.py
sep = "#"

class Solution:

    def encode(self, strs: List[str]) -> str:
        retString = ""
        for string in strs:
            retString+=(str(len(string))+sep+string)

        print(retString)
        return retString

    def decode(self, s: str) -> List[str]:
        retArray = []

        i = 0

        while i < len(s):
            j = i
            print(s, j)
            while s[j] != sep:
                j+=1
            strLen = int( s[i:j] )
            retArray.append(s[(j+1) : j+1+strLen])
            
            i = j+1+strLen
        


        return retArray

File: Data Structures & Algorithms/anagram-groups/submission-4.swift
class Solution {
    func groupAnagrams(_ strs: [String]) -> [[String]] {
        var dict: [[Character:Int]:[String]] = [:]

        for s in strs {
            var tempDict: [Character:Int] = [:]
            for c in s {
                tempDict[c, default: 0] += 1
            }
            dict[tempDict, default: []].append(s)

        }
        return Array(dict.values)
    }
}
